packages/token/index.ts

```diff
@@ -1,18 +1,98 @@
-import type { HexString } from "~typing"
-
-export class Token {
-  public constructor(
-    public readonly address: HexString,
-    public readonly symbol: string,
-    public readonly decimals: number
-  ) {}
+import { Contract, Interface } from "ethers"
+
+import type { ContractRunner } from "~packages/node"
+import type { BigNumberish, HexString } from "~typing"
+
+export type Token = {
+  address: HexString
+  symbol: string
+  decimals: number
+}
+
+export const ETH: Token = {
+  address: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
+  symbol: "ETH",
+  decimals: 18
 }
 
-// TODO: Network specific
-export const ETH = new Token(
-  "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
-  "ETH",
-  18
-)
+export class TokenContract {
+  public readonly address: HexString
```

Comment:
It could be better to call it `ERC20Contract`, and put it to `packages/contract/erc20.ts`